src/storage/src/hummock/sstable/block_iterator.rs

```diff
@@ -166,12 +169,17 @@ impl BlockIterator {
             self.decode_prefix_at(offset, self.last_key_len_type, self.last_value_len_type);
         self.key.truncate(prefix.overlap_len());
         self.key
-            .extend_from_slice(&self.block.data()[prefix.diff_key_range()]);
+            .extend_from_slice(self.block.slice(prefix.diff_key_range()));
 
         self.value_range = prefix.value_range();
         self.offset = offset;
         self.entry_len = prefix.entry_len();
 
+        self.hitmap.fill_with_ratio(
```

Comment:
May use fraction instead of f64, or we pass (start, end, len) of usize. May increase unnecessary cost on floating point computation for each key.